components/modal/Calendar/CalendarModal.tsx

```diff
@@ -0,0 +1,108 @@
+import moment from "moment-timezone"
+import { useCallback, useEffect, useRef, useState } from "react"
+import { LogService } from "../../../../core/LogService";
+import { CalendarProps } from "../../fields/datePicker/DatePickerField";
+
+interface CalendarModalProps extends CalendarProps {
+    readonly onChangeCallback?: (value: string) => void;
+    readonly calendarStyling?: (day: moment.Moment) => "" | "before" | "today";
+    readonly focus?: (value: boolean) => void;
+}
+
+interface WeekMoment {
+    value: moment.Moment;
+    index: number;
+    array?: moment.Moment[];
+}
+
+const LOG = LogService.createLogger('CalendarModal');
+
+
+export function Calendar(props: CalendarModalProps) {
```

Comment:
The name of this component should be `CalendarModal` if the name of the file is `CalendarModal.tsx`